pkg/koordlet/resmanager/cpu_suppress.go

```diff
@@ -182,29 +176,37 @@ func calculateBESuppressCPUSetPolicy(cpusetQuantity *resource.Quantity, oldCPUSe
 	// NOTE: here we prioritize the processors by the affinity of numa node/socket and hyper-threading so that the
 	//       cpuset cpus can be picked in order, rather than scoring all the possible policies.
 	// TBD: may just use a lookup table for a steady state
-
-	// initially sorted by processor id
-	prioritizedCPUs := nodeCPUInfo.ProcessorInfos
+	prioritizedCPUs := processorInfos
 	sort.Slice(prioritizedCPUs, func(i, j int) bool {
 		return prioritizedCPUs[i].CPUID < prioritizedCPUs[j].CPUID
 	})
 
 	// getNodeIndex is a function to calculate an index for every numa node or socket
 	getNodeIndex := func(info util.ProcessorInfo) int32 {
+		// (nodeId, socketId) => nodeIndex
 		return (info.NodeID + numProcessors) * (info.SocketID + 1)
 	}
 
 	// get the relative indexes of the processor on its node
-	indexOfNode := map[int32]int32{}
-	numProcessorsOfNode := map[int32]int32{}
+	coreIndexOfNode := map[int32]int32{}
+	coreIdSetofNode := map[int32]map[int32]int32{}
 	for _, processor := range prioritizedCPUs {
-		numProcessorsOfNode[getNodeIndex(processor)]++
-		if indexOfNode[processor.CoreID] <= 0 {
-			indexOfNode[processor.CoreID] = numProcessorsOfNode[getNodeIndex(processor)]
+		nodeIndex := getNodeIndex(processor)
+		if _, ok := coreIdSetofNode[nodeIndex]; !ok {
+			coreIdSetofNode[nodeIndex] = map[int32]int32{}
 		}
+		coreIdSetofNode[nodeIndex][processor.CoreID]++
 	}
 
-	// first we prefer the scattered node/socket, secondly the same core
+	for _, processor := range prioritizedCPUs {
+		coreIndexOfNode[processor.CoreID] = int32(len(coreIdSetofNode[getNodeIndex(processor)]))
```

Comment:
In the past, I have encountered a special machine with the same Core ID under different Sockets. Should not directly take the CoreID to participate in the calculation.

Reply:
@hormes reminded that the data of `lscpu -e` and `/proc/cpuinfo` are inconsistent. After investigation and analysis, the problems I encountered in the past should be caused by the internal version not processing the data of cadvisor correctly. Taking the X86 architecture as an example, the so-called `CoreID` we see through `lscpu -e` is actually the minimum ID among the two logical cores under the same physical core. For example, Core A has two logical cores 26 and 80, then the CoreID is 26.